dynamic_config/dynamic_config_manager_test.py

```diff
@@ -1,20 +1,21 @@
 """Tests for dynamic_config_manager."""
 
 import dynamic_config_manager
-from nighthawk.api.configuration import cluster_config_manager_pb2
+from nighthawk.api.configuration import cluster_config_manager_pb2, endpoints_config_manager_pb2
+from envoy.config.endpoint.v3 import endpoint_pb2
 from unittest import mock
 import pytest
 import random
 
 from envoy.service.discovery.v3 import discovery_pb2
 
 
-def toggle_add_cluster(choices):
-  """Toggle random result deterministically starting with True."""
-  if not hasattr(toggle_add_cluster, "add_cluster"):
-    toggle_add_cluster.add_cluster = True
-  ret = toggle_add_cluster.add_cluster
-  toggle_add_cluster.add_cluster = not ret
+def toggle_bool(choices):
+  """Use to deterministically patch random choice when initially assigning resources, starting with True."""
+  if not hasattr(toggle_bool, "next_value"):
```

Comment:
one other thing -- this function is very confusing to me. I don't understand what the param choices is doing, as it doesn't seem to be used? And are we checking for attributes on the function itself?

Is there a way to make this function work with less what I assume is python magic, to improve readability?

Reply:
sorry about that, simplied it.